design-proposals/klip-34-optional-with-in-c-star-stmts.md

```diff
@@ -0,0 +1,220 @@
+
+* Addition of new server configuration to provide a default for value format.
+* Removal of requirement for `VALUE_FORMAT` property in the `WITH` clause of CT/CS statements, where
+  the server configuration provides a default.
+* Removal of requirement for `KAFKA_FORMAT` property in the `WITH` clause of CT/CS statements.
+
+## What is not in scope
+
+Anything and everything else =)
+
+## Value/Return
+
+Clearer and cleaner syntax, with less unnecessary typing. Lowers barrier of entry for new users who
+are already familiar with Ansi SQL.
+
+## Public APIS
+
+### Server configs
+
+The following new configuration options will be added. These configurations can be set globally, 
+within the application property file, or locally, via the `SET` command.
+
+* `ksql.persistence.default.format.value`: the default value format. Default: none.
+
+* `ksql.persistence.topic.name.case`: if `lower`, defaults ksql to using lowercase topics names
+  where topic names are not explicitly provided. This includes matching a `CREATE TABLE` or 
+  `CREATE STREAM` statement to an existing topic, and includes the case of sink topics created
+  by `CREATE STREAM AS SELECT` and `CREATE TABLE AS SELECT`. Default: `upper`.
+
+* `ksql.persistence.topic.name.case.insensitive`: if `true`, matching a `CREATE TABLE` or 
+  `CREATE STREAM` statement to an existing source topic, and matching `CREATE TABLE AS SELECT` and
+  `CREATE STREAM AS SELECT` to an existing sink topic, will be case insensitive. Multiple matches 
+  will result in an error. Default: `false`.
+  
+Optionally:
+
+* `ksql.persistence.topic.partitions`: if set, the number of partitions to use when creating a topic.
+  If the user issues a `CREATE TABLE` or `CREATE STREAM` and the topic does not already exist, then
+  ksqlDB will create the topic with this number of partitions.   
+  
+  This introduces the potential for confusion, should the user mistype the source or topic name and
+  ksqlDB creates a new topic, rather than using an existing topic. However, as users would need to
+  opt-in by setting this config, we think this is acceptable.
+
+
+### CREATE properties
+
+The following properties will not longer be required in the `WITH` clause of `CREATE TABLE` and 
+`CREATE STREAM` statements, though the user may still choose to supply them.
+ 
+| `WITH` clause property |  Configuration required for property to be optional |
+|------------------------|-----------------------------------------------------|
+| `KAFKA_TOPIC`          | N/A                                                 |
+| `VALUE_FORMAT`         | `ksql.persistence.default.format.value`             |
+
+## Design
+
+### Optional `KAFKA_TOPIC`
+
+We propose changing the `KAFKA_TOPIC` property from _required_ to _optional_ in `CREATE STREAM` and 
+`CREATE TABLE` statements. If not supplied, it will default to the uppercase name of the stream or 
+table. This matches the behaviour of the property in `CREATE STREAM AS SELECT` and 
```

Comment:
This is not true. ksqlDB defaults to using the case-sensitive name of the stream or table, not the uppercase name. See comments above.